src/solvers/dgsem_tree/dg_2d_parabolic.jl

```diff
@@ -500,6 +510,367 @@ function calc_boundary_flux_by_direction_divergence!(surface_flux_values::Abstra
+    @unpack surface_flux = surface_integral
+    @unpack u_lower, u_upper, orientations = cache.mortars
+    @unpack fstar_upper_threaded, fstar_lower_threaded = cache
+
+    @threaded for mortar in eachmortar(dg, cache)
+        # Choose thread-specific pre-allocated container
+        fstar_upper = fstar_upper_threaded[Threads.threadid()]
+        fstar_lower = fstar_lower_threaded[Threads.threadid()]
+
+        # Calculate fluxes
+        orientation = orientations[mortar]
+        calc_fstar!(fstar_upper, equations_parabolic, surface_flux, u_transformed, dg, u_upper, mortar,
+                    orientation, cache_parabolic)
+        calc_fstar!(fstar_lower, equations_parabolic, surface_flux, u_transformed, dg, u_lower, mortar,
+                    orientation, cache_parabolic)
+
+        mortar_fluxes_to_elements!(surface_flux_values,
+                                   mesh, equations_parabolic, mortar_l2, dg, cache,
+                                   mortar, fstar_upper, fstar_lower)
+    end
+
+    return nothing
+end
+
+@inline function calc_fstar!(destination::AbstractArray{<:Any, 2}, 
+                             equations_parabolic::AbstractEquationsParabolic,
+                             surface_flux, dg::DGSEM,
+                             u_interfaces, interface, orientation,
+                             cache_parabolic)
+    for i in eachnode(dg)
+        # Call pointwise two-point numerical flux function
+        u_ll, u_rr = get_surface_node_vars(cache_parabolic.interfaces.u, equations_parabolic, dg, i, interface)
+        #flux = surface_flux(u_ll, u_rr, orientation, equations)
+        # TODO: parabolic; only BR1 at the moment
+        flux = 0.5 * (u_ll + u_rr) # CARE: Not sure if correct!
+
+        # Copy flux to left and right element storage
+        set_node_vars!(destination, flux, equations_parabolic, dg, i)
+    end
+
+    return nothing
+end
+
+@inline function calc_fstar!(destination::AbstractArray{<:Any, 2}, 
+                             equations_parabolic::AbstractEquationsParabolic,
+                             surface_flux, 
+                             u_transformed::AbstractArray,
+                             dg::DGSEM,
+                             u_interfaces, interface, orientation,
+                             cache_parabolic)                      
+    for i in eachnode(dg)
+        # Call pointwise two-point numerical flux function
+        
+        # TODO: Transformed u or regular u? Regular u is used in "calc_gradient!" -> "interface flux"
+        #u_ll, u_rr = get_surface_node_vars(u_transformed, equations_parabolic, dg, i, interface)
+        u_ll, u_rr = get_surface_node_vars(cache_parabolic.interfaces.u, equations_parabolic, dg, i, interface)
+
+        # CARE: Not sure if correct!
+        #flux = surface_flux(u_ll, u_rr, orientation, equations_parabolic)
+        # TODO: parabolic; only BR1 at the moment
+        flux = 0.5 * (u_ll + u_rr)
+
```

Comment:
**Question:** 

So this is the routine called during `calc_gradient`. I am not sure whether I should use `u_transformed` or `cache_parabolic.interfaces.u` at the surface node retrieval.

My initial guess was that I would use `u_transformed` since these are also the values prolonged to the boundaries:

https://github.com/DanielDoehring/Trixi.jl/blob/0bbe913647b05d6928e54335ca35392a079d292c/src/solvers/dgsem_tree/dg_2d_parabolic.jl#L949-L953

However, in the `interface_flux` computation, one is using `cache_parabolic.interfaces.u`

https://github.com/DanielDoehring/Trixi.jl/blob/0bbe913647b05d6928e54335ca35392a079d292c/src/solvers/dgsem_tree/dg_2d_parabolic.jl#L917-L938

So I am a bit confused here.

Reply:
The choice between `u_transformed` and `cache_parabolic.interfaces.u` has actually a quite noticeable difference:

<details> 
  <summary>u_transformed </summary>
  
![u_transformed](https://github.com/trixi-framework/Trixi.jl/assets/75639095/5937fdf0-1924-4103-875d-696d7fbe1510)

</details>

<details> 
  <summary>u </summary>

![u](https://github.com/trixi-framework/Trixi.jl/assets/75639095/e18b3764-710b-443b-a5fa-f3d41605b393)

</details>

Reply:
I believe you should use `cache_parabolic.interfaces.u`. 

`u_transformed` are the nodal values transformed to the variables that we take the gradient of, e.g., primitive or entropy variables. `cache_parabolic.interfaces.u` should store the prolonged values of `u_transformed`, which we then use to compute interface fluxes. Does that help?

Reply:
Thanks a lot, I found a different bug, things look now already quite well!